src/code/z_player_lib.c

```diff
@@ -176,11 +1151,37 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801271B0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80127438.s")
+// Player_SetMaskSegment?
+s32 func_80127438(PlayState* play, Player* player, s32 currentMask) {
+    if ((player->maskObjectLoadState == 0) && (currentMask == (u8)player->maskId)) {
+        OPEN_DISPS(play->state.gfxCtx);
+
+        gSPSegment(POLY_OPA_DISP++, 0x0A, player->maskObjectSegment);
+
+        CLOSE_DISPS(play->state.gfxCtx);
+
+        return true;
+    }
+
+    return false;
+}
+
+void func_80127488(PlayState* play, Player* player, u8 alpha) {
+    OPEN_DISPS(play->state.gfxCtx);
+
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gDPSetEnvColor(POLY_XLU_DISP++, 255, 0, 0, alpha);
+    gSPDisplayList(POLY_XLU_DISP++, gLinkGoronGoronPunchEffectDL);
+
+    func_80122BA4(play, &player->unk_3D0, 3, alpha);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80127488.s")
+    CLOSE_DISPS(play->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_8012754C.s")
+void func_8012754C(PlayState* play, Player* player) {
+    gSegments[0xA] = PHYSICAL_TO_VIRTUAL(player->maskObjectSegment);
+    AnimatedMat_DrawOpa(play, Lib_SegmentedToVirtual(&object_mask_meoto_Matanimheader_001CD8));
+}
```

Comment:
`DrawCouplesMask`, I suppose, probably should be a bit more specific.

Reply:
The item enum is `ITEM_MASK_COUPLE`, so shouldn't it be `Player_DrawMaskCouple`?

Reply:
It's not going to come up in a search anyway, and I intend to move the item names over to be more like the actual item names soon anyway.